File: ruby/hamming/hamming.rb
class Hamming
  VERSION=1
  def self.compute(a, b)
    raise ArgumentError if a.length != b.length
    diff = 0
    (0...a.length).each do |i|
      diff += 1 unless a[i] == b[i]
    end
    diff
  end
end

File: scala/hamming/src/main/scala/Hamming.scala
/**
  * Created by manmeetsinghsaluja on 17/5/16.
  */
object Hamming {
  def compute(str1: String, str2: String): Int ={
    var diff = 0
    if(str1.length != str2.length){
      throw new IllegalArgumentException("Length of both strings should be equal")
    }
    var counter = 0
    for (c <- str1){
      if(!c.equals(str2.charAt(counter))) diff += 1
      counter += 1
    }

    return diff
  }
}
